libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/audioguidance/CarAudioGuidanceAction.kt

```diff
@@ -5,24 +5,45 @@ import androidx.car.app.Screen
 import androidx.car.app.model.Action
 import androidx.car.app.model.CarIcon
 import androidx.core.graphics.drawable.IconCompat
-import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.coroutineScope
-import androidx.lifecycle.repeatOnLifecycle
 import com.mapbox.androidauto.R
-import com.mapbox.androidauto.action.MapboxActionProvider
 import com.mapbox.navigation.ui.voice.api.MapboxAudioGuidance
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.take
 import kotlinx.coroutines.launch
 
 /**
  * This class creates an action that can control audio guidance.
  */
-class CarAudioGuidanceUi : MapboxActionProvider {
+class CarAudioGuidanceAction {
+
     /**
-     * Android auto action for enabling and disabling the car navigation.
-     * Attach this to the screen while navigating.
+     * Build the [Action].
      */
+    fun getAction(screen: Screen): Action {
+        screen.invalidateOnceAfterStateChange()
+        return buildSoundButtonAction(screen)
+    }
+
+    // Actions are built but they are not destroyed, so the only way to know if an action is
+    // destroyed is if the Screen host is destroyed. For the AudioGuidanceAction, we also assume
+    // that there only needs to be one state change listener at a time.
+    private fun Screen.invalidateOnceAfterStateChange() {
+        invalidatorJob?.cancel()
+        invalidatorJob = lifecycle.coroutineScope.launch {
+            MapboxAudioGuidance.getRegisteredInstance().stateFlow()
+                .distinctUntilChanged { old, new ->
+                    old.isMuted == new.isMuted && old.isPlayable == new.isPlayable
+                }
+                .drop(1)
+                .take(1)
+                .collect { invalidate() }
+        }
+    }
+
```

Comment:
I'm considering this an issue related to "Customizable Actions". Considering that Actions do not have a lifecycle, it seems it is the `Screen` responsibility to know when to update state. 

So this "invalidate on state change" responsibility could arguably be moved to the `ActiveGuidanceScreen`. I chose to leave it here, so that the Action could be easily ported to any Screen.

I think the consequence here is that, there could be unnecessary calls to `Screen.invalidate()`. Specifically when you decide to remove the CarAudioGuidanceAction after adding it to a screen. But that is a very small consequence so I figured it okay.